waku/waku_peer_exchange/protocol.nim

```diff
@@ -129,11 +130,19 @@ proc getEnrsFromCache(
   return shuffledCache[0 ..< min(shuffledCache.len.int, numPeers.int)]
 
 proc populateEnrCache(wpx: WakuPeerExchange) =
-  # share only peers that i) are reachable ii) come from discv5
-  let withEnr = wpx.peerManager.peerStore
-    .getReachablePeers()
-    .filterIt(it.origin == Discv5)
-    .filterIt(it.enr.isSome)
+  # share only peers that i) are reachable ii) come from discv5 iii) share cluster
+  let withEnr = collect(newSeq):
+    for peer in wpx.peerManager.peerStore.getReachablePeers():
+      if peer.origin != Discv5:
+        continue
+
+      if peer.enr.isNone():
+        continue
+
+      if wpx.cluster.isSome() and peer.enr.get().isClusterMismatched(wpx.cluster.get()):
+        continue
```

Comment:
Maybe we can add trace logs under this conditions to help debugging in the future